docs(ssr): update middlewareMode comment (#5959)
vitejs/vite: docs/guide/ssr.md

  // Create Vite server in middleware mode. This disables Vite's own HTML
  // serving logic and let the parent server take control.
  //

>   // If you want to use Vite's own HTML serving logic, use 'html' as a middlewareMode
>   // (It will use Vite as a development middleware).

Reviewer: What about?
```suggestion
  // In middleware mode, if you want to use Vite's own HTML serving logic
  // use `'html'` as the `middlewareMode`.
```

Author: It might be a bit unclear what "In middleware mode" means. "In middleware mode" is more an implementation detail right? It might not be that relevant.

"If you want to use Vite's own HTML serving logic" feels more important because it mentions when to set the value to `html`

We can even hide the middleware mode implementation detail since anyway the variable name `middlewareMode` is explanatory
```suggestion
  // If you want Vite to handle HTML serving logic, set `middlewareMode` to `'html'`
```

Or maybe we can explain what we mean by `middlewareMode: 'html'`-
```suggestion
  // If you want Vite to handle HTML serving logic, set `middlewareMode` to `'html'`
  // (It will add Vite's HTML handler middleware)
```

Reviewer: It isn't an implementation detail, it is important in this context. You can run Vite in two modes. Normally it will take care of the server and there is HTML logic handling by default. But if you already have a server, you can run it in Middleware Mode so you can use Vite's middleware in your own server. In this mode, normally you won't need the HTML logic but there are some cases where it is handy to keep it. See https://vitejs.dev/config/#server-middlewaremode

Author: Ohh understood. Yup, then the above suggestion makes more sense. We can add this link https://vitejs.dev/config/#server-middlewaremode in brackets there. That link explained it well. Will make the changes and let you know.

Author: done. let me know if this looks fine. 038896e14ad223ec80b4150dfef6e21c49ed5d36
